starlite/contrib/sqlalchemy/repository.py

```diff
@@ -0,0 +1,540 @@
+        - For backends supporting `RETURNING` with `executemany`, a single bulk insert with returning clause is executed.
+        - For other backends, it does a bulk insert and then selects the inserted records
+
+        Args:
+            data: A list of instances to update.  Each should have a value for `self.id_attribute` that exists in the
+                collection.
+
+        Returns:
+            The updated instances.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found with same identifier as `data`.
+        """
+        data_to_update: list[dict[str, Any]] = [v.to_dict() if isinstance(v, self.model_type) else v for v in data]  # type: ignore
+        with wrap_sqlalchemy_exception():
+            if self.session.bind.dialect.update_executemany_returning:
+                instances = list(
+                    await self.session.scalars(  # type: ignore
+                        update(self.model_type).returning(self.model_type),
+                        data_to_update,  # pyright: reportGeneralTypeIssues=false
+                    )
+                )
+                await self.session.flush()
+                for instance in instances:
+                    self.session.expunge(instance)
+                return instances
+            updated_primary_keys: list[UUID] = []
+            for datum in data_to_update:
+                updated_primary_keys.append(datum.get("id"))  # type: ignore[arg-type]
+
+            await self.session.execute(
+                update(self.model_type),
+                data_to_update,
+            )
+            await self.session.flush()
+            # select the records we just updated.
+            return await self.list(CollectionFilter(field_name="id", values=updated_primary_keys))
+
+    async def list_and_count(
+        self,
+        *filters: FilterTypes,
+        **kwargs: Any,
+    ) -> tuple[list[ModelT], int]:
+        """List records with total count.
+
+        Args:
+            *filters: Types for specific filtering operations.
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            Count of records returned by query, ignoring pagination.
+        """
+        statement = self.statement.add_columns(
+            over(
+                sql_func.count(  # pylint: disable=not-callable
+                    self.model_type.id,  # type:ignore[attr-defined]
+                ),
+            )
+        )
+        statement = self._apply_filters(*filters, statement=statement)
+        statement = self._filter_select_by_kwargs(statement, **kwargs)
+        with wrap_sqlalchemy_exception():
+            result = await self._execute(statement)
+            count: int = 0
+            instances: list[ModelT] = []
+            for i, (instance, count_value) in enumerate(result):
+                self.session.expunge(instance)
+                instances.append(instance)
+                if i == 0:
+                    count = count_value
```

Comment:
```suggestion
            first, count = next(result)
            instances = [first]
            for instance, _ in result:
                self.session.expunge(instance)
                instances.append(instance)
```
I think this should work and looks a bit cleaner, haven't tested though  :worried: